### ISSUE TITLE
Fix color contrast in search results

### ISSUE THREAD

**Issue by atsansone** [dart-lang/site-www#5794]
Hotlist: https://buganizer.corp.google.com/hotlists/5891206
Bug: https://buganizer.corp.google.com/issues/338014334

### Observed Results

Some links and text in the page do not have minimum colour contrast ratio of 4.5/1.

* [Screencast](https://screencast.googleplex.com/cast/NDcwODc1MDczOTg5ODM2OHwwNWU2ZDMzZi1kMw).
* [Screencast](http://screencast/cast/NjIyMzQ2Njk0NzY3NDExMnw5Y2UzODUxMC1hZQ)

### Expected Results

The content of the page should have a minimum colour contrast ratio of 4.5/1.

### User Impact

Users who use forced colors will not be able to read or interact with the content which does not have minimum colour contrast ratio which will lead to loss of context. This will effect users with Low vision, colour blindness.

### Steps To Reproduce

1. Enable High contrast mode.
1. Go to https://dart.dev/search?cx=011220921317074318178%3A_yy-tmb5t_i&ie=UTF-8&hl=en&q=isolate.
1. Using the <kbd>Tab</kbd> key, navigate to "search" and type "isolate" and search.
1. Navigate through the search results and observe contrast.

### Relevent GAR Sheet

[A11Y Tracker](https://a11ytracker.corp.google.com/products/034ba691-ac78-4cc2-88c2-e06449cfd17e/edit/testresults?useCase=7jg80pq64n0p8&e=ProductLabels).

### GAR Details

[1.14. Is there sufficient color contrast between the text/icons and the background?](http://go/gar-web-color-contrast)

### Additional Info

A second video has been attached since there several elements that have the same color contrast issue
in the following https://dart.dev/codelabs/async-await). This in order to don't make a duplicate bu.

### Environment Details

* Machine: Chromebook.
* Browser: Chrome (123.0.6312.112 (Official Build) (64-bit)).
* Accessibility Tool: High contrast mode.
* Bug Filling Time: 10 mins.

**Comment by parlough** [comment]
These search results are generated, styled, and managed by the Google Custom Search setup we use.

Since we don't control that, I'm going to close this in favor of an umbrella issue https://github.com/dart-lang/site-www/issues/6486 which tracks implementing a new experience that we can properly integrate and style. Thanks!
